.github/workflows/go-foundation.yml

```diff
@@ -0,0 +1,28 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.20
+
+    - name: Build
+      run: go build -v ./...
```

Comment:
Later when we have our server and compiler we'll change this to the main server file?

Reply:
We should change this to whatever are our "main" files once we have it, yup (and that would be the entrypoint of our server as a binary).
I'm not sure we want to stick to this for building, even... But it'd still make sense for running the tests.
I'd suggest we use a Makefile in the future (although, from my limited knowledge with gh actions, it gets more annoying to set it up to call a make command with all proper dependencies).